CenterForOpenScience/osf.io
# [hotfix] Add explicit ToS checkbox to signup pages [PLAT-841][PLAT-842]

❗️**This PR cannot be released until our Terms of Use and Privacy Policy have actually been updated**❗️ 

#### Purpose
- Supports GDPR compliance by requiring explicit consent to our Terms of Use and Privacy Policy from new and existing users.
  - Records the time when the user agreed to our updated terms of service.

#### Changes
* Adds an `accepted_terms_of_service` datetime field to the `User` model.
* Adds an `accepted_terms_of_service` write-only BooleanField to the user serializer (can only be set in put/patch requests).
* Adds "I have read and agree to..." checkboxes to the register, claim account, and ORCID (external) log-in/sign-up pages. 
* Adds a terms of use banner/status message to be displayed to existing users that need to agree to our updated terms.

#### QA Notes
* "Terms of Use" and "Privacy Policy" should always be linked. 
* Check ToS banner interactions with other banners (it probably won't be beautiful, but nothing should break)
* Register page:
  * Submit button should be disabled until terms box has been checked. 
* ORCID log-in/sign-up page:
  * Send button is not disabled, but if you click without checking the box you'll see an error popup.
* Claim account page: 
  * Save button should be disabled until the checkbox is checked.
* The banner for existing users can be dismissed, but will reappear on the next page view if the user didn't agree. If they've agreed, they should never see the banner again.
  * Continue button should be disabled until the checkbox is checked.

![final_banner_project_page](https://user-images.githubusercontent.com/7913604/40135205-ba1bc852-5912-11e8-879a-4f54d58e47f3.png)



#### Ticket
- [PLAT-841](https://openscience.atlassian.net/browse/PLAT-841)
